src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -5185,12 +5213,6 @@ int32_t NaClSysAccept(struct NaClAppThread *natp,
   NaClLog(2, "Cage %d Entered NaClSysAccept(0x%08"NACL_PRIxPTR", %d, 0x%08"NACL_PRIxPTR", 0x%08"NACL_PRIxPTR")\n",
           nap->cage_id, (uintptr_t) natp, sockfd, (uintptr_t) addr, (uintptr_t) addrlen);
 
-  nd = malloc(sizeof(struct NaClHostDesc));
-  if (!nd) {
-    NaClLog(2, "NaClSysAccept could not allocate room for returning NaCl desc\n");
-    return -NACL_ABI_ENOMEM;
-  }
-
```

Comment:
Now that you have moved the allocation of nd down, you cannot have in every error out below this `free(nd)`. I would be a proponent of moving the allocation back up here because IMO the earlier we can handle an ENOMEM the better.

Reply:
The alternative is of course just to remove the frees below this

Reply:
Id rather allocate after lind_accept since we dont need to make a descriptor if its an error.

Reply:
fixed